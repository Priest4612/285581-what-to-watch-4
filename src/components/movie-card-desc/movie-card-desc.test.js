import React from 'react';
import renderer from 'react-test-renderer';

import MovieCardDesc from './movie-card-desc.jsx';

import filmsMock from '../../mocks/test-films-mock.json';


it(`Render MovieCardDesc`, () => {
  const {name, genre, released} = filmsMock[0];
  const tree = renderer
    .create(<MovieCardDesc
      name={name}
      genre={genre}
      released={released}
    >
      <a>la-la-la</a>
    </MovieCardDesc>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
