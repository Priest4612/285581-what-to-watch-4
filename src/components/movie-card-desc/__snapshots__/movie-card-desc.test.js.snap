// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieCardDesc -- full 1`] = `
Array [
  <div
    className="movie-card__desc"
  >
    <h2
      className="movie-card__title"
    >
      Macbeth
    </h2>
    <p
      className="movie-card__meta"
    >
      <span
        className="movie-card__genre"
      >
        Мишель Родригес
      </span>
      <span
        className="movie-card__year"
      >
        1962
      </span>
    </p>
  </div>,
  <div
    className="movie-card__buttons"
  >
    <button
      className="btn btn--play movie-card__button"
      type="button"
    >
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
      >
        <use
          xlinkHref="#play-s"
        />
      </svg>
      <span>
        Play
      </span>
    </button>
    <button
      className="btn btn--list movie-card__button"
      type="button"
    >
      <svg
        height="20"
        viewBox="0 0 19 20"
        width="19"
      >
        <use
          xlinkHref="#add"
        />
      </svg>
      <span>
        My list
      </span>
    </button>
    <a
      className="btn movie-card__button"
      href="add-review.html"
    >
      Add review
    </a>
  </div>,
]
`;

exports[`Render MovieCardDesc -- main 1`] = `
Array [
  <div
    className="movie-card__desc"
  >
    <h2
      className="movie-card__title"
    >
      Bohemian Rhapsody
    </h2>
    <p
      className="movie-card__meta"
    >
      <span
        className="movie-card__genre"
      >
        Роуэн Эткинсон
      </span>
      <span
        className="movie-card__year"
      >
        1930
      </span>
    </p>
  </div>,
  <div
    className="movie-card__buttons"
  >
    <button
      className="btn btn--play movie-card__button"
      type="button"
    >
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
      >
        <use
          xlinkHref="#play-s"
        />
      </svg>
      <span>
        Play
      </span>
    </button>
    <button
      className="btn btn--list movie-card__button"
      type="button"
    >
      <svg
        height="20"
        viewBox="0 0 19 20"
        width="19"
      >
        <use
          xlinkHref="#add"
        />
      </svg>
      <span>
        My list
      </span>
    </button>
  </div>,
]
`;
